node/lib/cmd/stash.js

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2017, Two Sigma Open Source
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of git-meta nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+"use strict";
+
+const co = require("co");
+
+/**
+ * This module contains the command entry point for stash.
+ */
+
+/**
+ * help text for the `stash` command
+ * @property {String}
+ */
+exports.helpText = `Stash changes to the index and working directory`;
+
+/**
+ * description of the `stash` command
+ * @property {String}
+ */
+exports.description =`
+Provide commands for saving and restoring the state of the monorepo.
+Note that 'stash' affects only the open repositories of *sub-repos*; as
+currenty implemented, the meta-repo itself is not affected, including staged
+and unstaged commits to the sub-repos.`;
+
+exports.configureParser = function (parser) {
+
+    parser.addArgument("type", {
+        help: "'save' to save a stash, 'pop' to restore; 'save' is default",
```

Comment:
I think you're wrong in this one though:

```
save [-p|--patch] [-k|--[no-]keep-index] [-u|--include-untracked] [-a|--all] [-q|--quiet] [<message>]
           Save your local modifications to a new stash, and run git reset --hard to revert them. The <message> part is optional and gives the description along
           with the stashed state. For quickly making a snapshot, you can omit both "save" and <message>, but giving only <message> does not trigger this action
           to prevent a misspelled subcommand from making an unwanted stash.
```
vs.
```
create
           Create a stash (which is a regular commit object) and return its object name, without storing it anywhere in the ref namespace. This is intended to be
           useful for scripts. It is probably not the command you want to use; see "save" above.
```